src/parse.js

```diff
@@ -245,11 +245,17 @@ export function parseFunction (str) {
 		let args = [];
 		let argMeta = [];
 		let lastAlpha = false;
+		let name = parts[1].toLowerCase();
 
 		let separators = parts[2].replace(regex.singleArgument, ($0, rawArg) => {
 			let { value, meta } = parseArgument(rawArg);
 
-			if ($0.startsWith("/")) {
+			if (
+				// If there's a slash here, it's modern syntax
+				$0.startsWith("/")
+				// If there's still elements to process after there's already 3 in `args` (and the we're not dealing with "color()"), it's likely to be a legacy color like "hsl(0, 0%, 0%, 0.5)"
+				|| (name !== "color" && args.length === 3)
```

Comment:
Yeah not super keen on that assumption, but the nature of legacy functional syntax makes it hard to avoid.
Having read the discussion I am in favor of merging it and logging the issue, as suggested.